src/TLuaInterpreter.cpp

```diff
@@ -10078,17 +10057,15 @@ int TLuaInterpreter::setMergeTables(lua_State* L)
 // Documentation: ? - public function missing documentation in wiki
 int TLuaInterpreter::pasteWindow(lua_State* L)
 {
-    string luaName;
+    QString window;
     if (!lua_isstring(L, 1)) {
-        lua_pushstring(L, "pasteWindow: wrong argument type");
-        lua_error(L);
-        return 1;
+        lua_pushfstring(L, "pasteWindow: bad argument #1 type (window name as string expected, got %s!)", luaL_typename(L, 1));
+        return lua_error(L);
     } else {
-        luaName = lua_tostring(L, 1);
+        window = QString::fromUtf8(lua_tostring(L, 1));
     }
     Host& host = getHostFromLua(L);
-    QString name(luaName.c_str());
-    mudlet::self()->pasteWindow(&host, name);
+    mudlet::self()->pasteWindow(&host, window);
```

Comment:
How does this handle the `main` `TConsole` case - or is it not possible to use `pasteWindow(...)` or does it mandate the use of `main` rather an empty string?  In a related manner does `mudlet::pasteWindow(...)` not deserve returning to the user with `true`/`nil + error message` as appropriate?

Reply:
Isn't it relevant to move text from one window to another (or a buffer) though?